test/parse_test.js

```diff
@@ -646,6 +646,18 @@ it('should fail on invalid examples', async function() {
   }, expectedErrorObject);
 });
 
+describe('memory usage', function () {
+  it('should use this same instance of validation function in each call', async function() {
+    const asyncapi = fs.readFileSync(path.resolve(__dirname, './good/asyncapi-complex-schema.yml'), 'utf8');
+
+    for (let i = 0, l = 1000; i < l; i++) {
+      await parser.parse(asyncapi);
+      const used = process.memoryUsage().heapUsed / 1024 / 1024;
+      expect(used < 60).to.equal(true); // less than 60 MB
```

Comment:
Should this not be tested outside of the for loop? 🤔 

Also, do you think this is reliable, in the long run? 🧐 What if other processes spikes during the test, etc? It would be rather vulnerable to false positives right?

Reply:
> Should this not be tested outside of the for loop? 🤔

No, because each call (without changes in PR) should increase memory and this test checks if memory usage is within a certain limit.

> Also, do you think this is reliable, in the long run? 🧐 What if other processes spikes during the test, etc? It would be rather vulnerable to false positives right?

`process.memoryUsage()` checks memory usage per process, so yes we can have the possibility that the memory usage will increase, that's why I gave this 60mb as +-10mb reserve - maximum usage is within 50mb +-1mb

Reply:
I don't really like this approach with checking memory, but any other solution would be too complex... Lets hope we will never see false positives 🤞